src/module/migration/migrations/912-rm-focus-trait-focus-cantrips.ts

```diff
@@ -0,0 +1,28 @@
+import { ItemSourcePF2e } from "@item/base/data/index.ts";
+import { MagicTradition } from "@item/spell/types.ts";
+import { MAGIC_TRADITIONS } from "@item/spell/values.ts";
+import * as R from "remeda";
+import { MigrationBase } from "../base.ts";
+
+/** Remove the "focus" traits from focus cantrips. */
+export class Migration912RmFocusTraitFocusCantrips extends MigrationBase {
+    static override version = 0.912;
+
+    override async updateItem(source: ItemSourcePF2e): Promise<void> {
+        if (source.type !== "spell") return;
+
+        const magicTraditions: Set<string> = MAGIC_TRADITIONS;
+        const traits = source.system.traits;
+        const misplacedMagicTraditions = traits.value.filter((t): t is MagicTradition => magicTraditions.has(t));
+        traits.traditions.push(...misplacedMagicTraditions);
+        traits.value = traits.value.filter((t) => !magicTraditions.has(t));
+
+        if (traits.value.includes("focus")) traits.traditions = [];
```

Comment:
This should maybe instead check if traditions are blank, rather than setting traditions to blank, and then remove the focus trait if so.